client/my-sites/plan-features-2023-grid/hooks/npm-ready/data-store/use-grid-plans.tsx

```diff
@@ -0,0 +1,301 @@
+	// API plans will be ported to data store and be queried from there
+	usePricedAPIPlans: UsePricedAPIPlans;
+	usePricingMetaForGridPlans: UsePricingMetaForGridPlans;
+	withoutProRatedCreditsForPricing?: boolean;
+	selectedFeature?: string | null;
+	term?: ( typeof TERMS_LIST )[ number ]; // defaults to monthly
+	intent?: PlansIntent;
+	selectedPlan?: PlanSlug;
+	sitePlanSlug?: PlanSlug | null;
+	hideEnterprisePlan?: boolean;
+	isInSignup?: boolean;
+	// whether plan is upgradable from current plan (used in logged-in state)
+	usePlanUpgradeabilityCheck?: ( { planSlugs }: { planSlugs: PlanSlug[] } ) => {
+		[ key: string ]: boolean;
+	};
+	showLegacyStorageFeature?: boolean;
+	// for AB Test experiment:
+	isGlobalStylesOnPersonal?: boolean;
+}
+
+const usePlanTypesWithIntent = ( {
+	intent,
+	selectedPlan,
+	sitePlanSlug,
+	hideEnterprisePlan,
+}: Pick< Props, 'intent' | 'selectedPlan' | 'sitePlanSlug' | 'hideEnterprisePlan' > ): string[] => {
+	const isEnterpriseAvailable = ! hideEnterprisePlan;
+	const isBloggerAvailable =
+		( selectedPlan && isBloggerPlan( selectedPlan ) ) ||
+		( sitePlanSlug && isBloggerPlan( sitePlanSlug ) );
+
+	let currentSitePlanType = null;
+	if ( sitePlanSlug ) {
+		currentSitePlanType = getPlan( sitePlanSlug )?.type;
+	}
+
+	const availablePlanTypes = [
+		TYPE_FREE,
+		...( isBloggerAvailable ? [ TYPE_BLOGGER ] : [] ),
+		TYPE_PERSONAL,
+		TYPE_PREMIUM,
+		TYPE_BUSINESS,
+		TYPE_ECOMMERCE,
+		...( isEnterpriseAvailable ? [ TYPE_ENTERPRISE_GRID_WPCOM ] : [] ),
+		TYPE_WOOEXPRESS_SMALL,
+		TYPE_WOOEXPRESS_MEDIUM,
+	];
+
+	let planTypes;
+	switch ( intent ) {
+		case 'plans-woocommerce':
+			planTypes = [ TYPE_WOOEXPRESS_SMALL, TYPE_WOOEXPRESS_MEDIUM ];
+			break;
+		case 'plans-blog-onboarding':
+			planTypes = [ TYPE_FREE, TYPE_PERSONAL, TYPE_PREMIUM, TYPE_BUSINESS ];
+			break;
+		case 'plans-newsletter':
+		case 'plans-link-in-bio':
+			planTypes = [ TYPE_FREE, TYPE_PERSONAL, TYPE_PREMIUM ];
+			break;
+		case 'plans-new-hosted-site':
+			planTypes = [ TYPE_FREE, TYPE_PERSONAL, TYPE_PREMIUM, TYPE_BUSINESS, TYPE_ECOMMERCE ];
+			break;
+		case 'plans-import':
+			planTypes = [ TYPE_FREE, TYPE_PERSONAL, TYPE_PREMIUM, TYPE_BUSINESS ];
+			break;
+		case 'plans-plugins':
+			planTypes = [
+				...( currentSitePlanType ? [ currentSitePlanType ] : [] ),
+				TYPE_BUSINESS,
+				TYPE_ECOMMERCE,
+			];
+			break;
+		case 'plans-jetpack-app':
+			planTypes = [ TYPE_PERSONAL, TYPE_PREMIUM ];
+			break;
+		case 'plans-default-wpcom':
+			planTypes = [
+				TYPE_FREE,
+				...( isBloggerAvailable ? [ TYPE_BLOGGER ] : [] ),
+				TYPE_PERSONAL,
+				TYPE_PREMIUM,
+				TYPE_BUSINESS,
+				TYPE_ECOMMERCE,
+				...( isEnterpriseAvailable ? [ TYPE_ENTERPRISE_GRID_WPCOM ] : [] ),
+			];
+			break;
+		default:
+			planTypes = availablePlanTypes;
+	}
+
+	return planTypes;
+};
+
+// TODO clk: move to plans data store
+const useGridPlans = ( {
+	usePricedAPIPlans,
+	usePricingMetaForGridPlans,
+	withoutProRatedCreditsForPricing,
```

Comment:
This prop is not currently wired up from plan-features-main. Probably related to the bug @southp discovered.

https://github.com/Automattic/wp-calypso/blob/b70413d3de75eb69c6bd453cdb510df6de45e16b/client/my-sites/plans-features-main/index.tsx#L333-L345

Reply:
I need to double-check, but as far as I understand it - it defaults to `false` on all occasions (from previous/`trunk`) except when we need the yearly variant plan's prices for comparison with monthly plan. So not passing anything when computing the grid plan prices sounds intentional.

Reply:
come to think of it - if that's the case, then might remove `withoutProRatedCreditsForPricing` altogether. It probably just lingered from piecing out the pricing extraction.